UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Services/FssService.cs

```diff
@@ -375,25 +376,85 @@ public async Task<IEnumerable<BatchFile>> GetAioInfoFolderFilesAsync(string batc
+
+            HttpResponseMessage httpReadMeFileResponse = await _fssApiClient.DownloadFile(readMeFilePath.TrimStart('/'), accessToken);
+
+            var requestUri = new Uri(httpReadMeFileResponse.RequestMessage.RequestUri.ToString()).GetLeftPart(UriPartial.Path);
+
+            if (httpReadMeFileResponse.IsSuccessStatusCode)
+            {
+                var serverValue = httpReadMeFileResponse.Headers.Server.ToString().Split('/').First();
+                await using Stream stream = await httpReadMeFileResponse.Content.ReadAsStreamAsync();
+                if (serverValue == ServerHeaderValue)
+                {
+                    _logger.LogInformation(EventIds.DownloadReadmeFile307RedirectResponse.ToEventId(), "File share service download readme.txt file redirected with uri:{requestUri} responded with 307 code for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", requestUri, batchId, correlationId);
+                }
+                return _fileSystemHelper.DownloadReadmeFile(filePath, stream);
+            }
+            else
+            {
+                _logger.LogError(EventIds.DownloadReadMeFileNonOkResponse.ToEventId(), "Error in file share service while downloading readme.txt file with uri:{requestUri} responded with {StatusCode} and BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", requestUri, httpReadMeFileResponse.StatusCode, batchId, correlationId);
+                throw new FulfilmentException(EventIds.DownloadReadMeFileNonOkResponse.ToEventId());
+            }
+        }
+
         //Private Methods
         [ExcludeFromCodeCoverage]
         private CreateBatchRequestModel CreateBatchRequestModel(Batch batchType)
         {
-
             string currentYear = DateTime.UtcNow.Year.ToString();
             string currentWeek = CommonHelper.GetCurrentWeekNumber(DateTime.UtcNow).ToString();
             CreateBatchRequestModel createBatchRequest;
 
             if (_aioBatchTypes.Contains(batchType))
             {
-                createBatchRequest = new()
+                createBatchRequest = new CreateBatchRequestModel
                 {
                     BusinessUnit = _fssApiConfiguration.Value.AioBusinessUnit,
                     ExpiryDate = DateTime.UtcNow.AddDays(28).ToString("yyyy-MM-ddTHH:mm:ss.fffZ", CultureInfo.InvariantCulture),
-                    Acl = new Acl()
+                    Acl = new Acl
                     {
-                        ReadUsers = string.IsNullOrEmpty(_fssApiConfiguration.Value.AioReadUsers) ? new() : _fssApiConfiguration.Value.AioReadUsers.Split(",").ToList(),
-                        ReadGroups = string.IsNullOrEmpty(_fssApiConfiguration.Value.AioReadGroups) ? new() : _fssApiConfiguration.Value.AioReadGroups.Split(",").ToList(),
+                        ReadUsers = string.IsNullOrEmpty(_fssApiConfiguration.Value.AioReadUsers) ? new List<string>() : _fssApiConfiguration.Value.AioReadUsers.Split(",").ToList(),
```

Comment:
Is 'ToList()' required at the end of lines 456 and 457? ReadUsers and ReadGroups are IEnumerables types

Reply:
True. ToList() are not required. 
Resolved.